### PR TITLE
i18n(pt-BR): Adding `partial-hydration.md` & `project-structure.md` 

### PR DESCRIPTION
**Pull request by Yan-Thomas**
Translations for `partial-hydration.md` and `project-structure.md`. I also made a small change and added new translations to nav.ts.

### CODE REVIEW

File: src/pages/pt-BR/core-concepts/partial-hydration.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+<!-- 100% HTML, Zero JavaScript! -->
+<MyReactComponent />
+```
+
+Porém as vezes, JavaScript no lado do cliente é necessário para criar interfaces de usuário interativas. Quando você se ver precisando de UI interativa na página, Astro não te força a utilizar JavaScript em 100% da página. Em vez disso, Astro utiliza uma técnica chamada **hidratação parcial** que te deixa hidratar componentes individualmente na página. Dessa forma, você envia apenas o JavaScript absolutamente necessário que você precisa para carregar a página.
+
+
+```astro
+---
+// Examplo: Use um componente React dinâmico na página.
```
Comment (caioferrarezi) [inline src/pages/pt-BR/core-concepts/partial-hydration.md:25]:
```suggestion
// Exemplo: Use um componente React dinâmico na página.
```

Comment (Yan-Thomas) [reply src/pages/pt-BR/core-concepts/partial-hydration.md:25]: Nice catch!

File: src/pages/pt-BR/core-concepts/partial-hydration.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+A maioria do seu site continua como puro e leve HTML e CSS, com isoladas **ilhas de interatividade**.
+
+## Hidratação Parcial
+
+Há vários casos onde você precisa de um componente de UI interativo que rode no navegador:
+
+- Um carrossel de imagens
+- Uma barra de pesquisa com autocompletação
```
Comment (caioferrarezi) [inline src/pages/pt-BR/core-concepts/partial-hydration.md:40]: What do you think about autocomplete being translated to `preenchimento automático`?

File: src/pages/pt-BR/core-concepts/partial-hydration.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+- Um botão para abrir/fechar uma barra lateral em dispositivos móveis
+- Um botão de "Compre Agora"
+
+No Astro, cabe a você como desenvolvedor escolher explicitamente quais componentes na página precisam rodar no navegador. Astro irá apenas hidratar exatamente o que é necessário na página e deixar o resto do seu site como HTML estático. Esta técnica é conhecida como **hidratação parcial**.
+
+**Hidratação parcial é o segredo da história de rapidez por padrão do Astro.**
+
+## Arquitetura em Ilhas
+
+**Arquitetura em ilhas** é a ideia de utilizar hidratação parcial para se construir websites inteiros. A arquitetura em ilhas é uma alternativa ao comum processo de build do seu website em um pacote de JavaScript no lado do cliente que precisa ser entregue ao usuário.
```
Comment (caioferrarezi) [inline src/pages/pt-BR/core-concepts/partial-hydration.md:50]:
```suggestion
**Arquitetura em ilhas** é a ideia de utilizar hidratação parcial para se construir websites inteiros. A arquitetura em ilhas é uma alternativa ao processo comum de construção do seu website em um pacote de JavaScript no lado do cliente que precisa ser entregue ao usuário.
```

File: src/pages/pt-BR/core-concepts/partial-hydration.md (hunk 2 above)
Comment (Yan-Thomas) [reply src/pages/pt-BR/core-concepts/partial-hydration.md:40]: I like it! It was the exact word I was looking for, thanks!

File: src/pages/pt-BR/core-concepts/partial-hydration.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+> A ideia geral de uma arquitetura em "ilhas" é ridiculamente simples: renderize páginas HTML no servidor, e injete placeholders ou slots em áreas altamente dinâmicas. <br/> -- [Arquitetura em Ilhas: Jason Miller](https://jasonformat.com/islands-architecture/)
+
+Apesar dos óbvios benefícios de performance de se enviar menos JavaScript para o navegador, há dois benefícios-chave para a arquitetura em ilhas:
+
+- **Componentes são carregados individualmente.** Um leve componente (como o toggle de uma barra lateral) será carregado e renderizado rapidamente sem ser bloqueado pelos componentes mais pesados na página.
+- **Componentes são renderizados isoladamente.** Cada parte da página é uma unidade isolada, e um problema de performance em uma unidade não irá diretamente afetar as outras.
+
+![diagram](https://res.cloudinary.com/wedding-website/image/upload/v1596766231/islands-architecture-1.png)
```
Comment (caioferrarezi) [inline src/pages/pt-BR/core-concepts/partial-hydration.md:59]: I think we should have a nice alt text here like (feel free to improve it): `Diagrama da estrutura de uma arquitetura em ilha, com seções de conteúdo estático e seções em destaque representando os componentes que serão hidratados, sendo eles o cabeçalho, o carossel de imagens e uma barra lateral`

File: src/pages/pt-BR/core-concepts/project-structure.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+Você pode colocar CSS e JavaScript no seu diretório `public/`, porém esteja atento que estes arquivos não serão empacotados ou otimizados na sua build final.
+
+ 💡 *Como uma regra geral, qualquer CSS ou JavaScript que você mesmo escrever deve estar no seu diretório `src/`.*
+
+### `package.json`
+
+Este é um arquivo utilizado por gerenciadores de pacotes de JavaScript para gerenciar suas dependências. Ele também define scripts que são comumente usados para rodar o Astro (ex: `npm start`, `npm run build`).
+
+Para ajuda em como criar um novo arquivo `package.json` para o seu projeto, veja as instruções de [instalação manual](/pt-BR/install/manual).
```
Comment (caioferrarezi) [inline src/pages/pt-BR/core-concepts/project-structure.md:101]: little typo, I think
```suggestion
Para ajudar em como criar um novo arquivo `package.json` para o seu projeto, veja as instruções de [instalação manual](/pt-BR/install/manual).
```

File: src/pages/pt-BR/core-concepts/partial-hydration.md (hunk 4 above)
Comment (Yan-Thomas) [inline src/pages/pt-BR/core-concepts/partial-hydration.md:59]:
```suggestion
![Diagrama da estrutura de uma arquitetura em ilha, com seções de conteúdo estático e seções em destaque representando os componentes que serão hidratados, sendo eles o cabeçalho, o carrossel de imagens e uma barra lateral](https://res.cloudinary.com/wedding-website/image/upload/v1596766231/islands-architecture-1.png)
```

Comment (caioferrarezi) [reply src/pages/pt-BR/core-concepts/partial-hydration.md:59]: Nice!

File: src/pages/pt-BR/core-concepts/project-structure.md (hunk 5 above)
Comment (Yan-Thomas) [reply src/pages/pt-BR/core-concepts/project-structure.md:101]: It was not technically a typo, but I think we should change the wording a bit to be clear about it:

```suggestion
Caso precise de ajuda em como criar um novo arquivo `package.json` para o seu projeto, veja as instruções de [instalação manual](/pt-BR/install/manual).
```

File: src/pages/pt-BR/core-concepts/partial-hydration.md (hunk 2 above)
Comment (Yan-Thomas) [inline src/pages/pt-BR/core-concepts/partial-hydration.md:40]:
```suggestion
- Uma barra de pesquisa com preenchimento automático
```

**Review by caioferrarezi — APPROVED** [review]
LGTM

**Pull request merged**